### PR TITLE
add support for zig

### PR DESCRIPTION
**Pull request by komuw**
Add support for Zig programming language's standard json library
- https://ziglang.org/

Screenshot 2021-02-26 at 15 42 48

### CODE REVIEW

File: parsers/test_zig/test_json.zig [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    const file = std.fs.cwd().openFile(file_name, .{}) catch |err| {
+        std.os.exit(1);
+    };
+    const reader = file.reader();
+    const buf = reader.readAllAlloc(allocator, std.math.maxInt(usize)) catch |err| {
+        std.os.exit(1);
+    };
+
+    var p = std.json.Parser.init(allocator, false);
+    var tree = p.parse(buf) catch |err| {
```
Comment (daurnimator) [inline parsers/test_zig/test_json.zig:23]: Avoid `std.json.Parser.parse` if you want control over json parsing, use `std.json.parse` instead.

Comment (komuw) [reply parsers/test_zig/test_json.zig:23]: @daurnimator `std.json.parse` takes a `type` as its first arg. However for this repo, the type of json string that will be handed to our program is only known at runtime; https://github.com/nst/JSONTestSuite/tree/master/test_parsing

**Comment by daurnimator** [comment]
Do you mean to commit a binary to the repository?

**Comment by komuw** [comment]
> Do you mean to commit a binary to the repository?

I followed what was already done for other languages, see Golang and C:     
- https://github.com/nst/JSONTestSuite/tree/master/parsers/test_go
- https://github.com/nst/JSONTestSuite/tree/master/parsers/test_json-c/bin

**Pull request merged**